eda/src/autogluon/eda/analysis/interaction.py

```diff
@@ -0,0 +1,111 @@
+from typing import List, Optional
+
+import phik  # noqa - required for significance_matrix instrumentation on pandas dataframes
+
+from .base import AbstractAnalysis
+from .. import AnalysisState
+
+__all__ = ["Correlation", "CorrelationSignificance"]
+
+
+class Correlation(AbstractAnalysis):
+    """
+    Correlation analysis.
+
+    Parameters
+    ----------
+    method: str  {'pearson', 'kendall', 'spearman', 'phik'}, default='spearman'
+        Method of correlation:
+            * pearson : standard correlation coefficient
+            * kendall : Kendall Tau correlation coefficient
+            * spearman : Spearman rank correlation
+            * phik : phi_k correlation
+                Correlation matrix of bivariate gaussian derived from chi2-value Chi2-value
+                gets converted into correlation coefficient of bivariate gauss with correlation
+                value rho, assuming giving binning and number of records. Correlation coefficient
+                value is between 0 and 1. Bivariate gaussian's range is set to [-5,5] by construction.
+                See Also `phik <https://github.com/KaveIO/PhiK>`_ documentation.
+
+    focus_field: Optional[str], default = None
+        field name to focus. Specifying a field would filter all correlations only when they are >= `focus_field_threshold`
+        This is helpful when dealing with a large number of variables.
+    focus_field_threshold: float, default = 0.5
+        a cut-off threshold when `focus_field` is specified
+    parent: Optional[AbstractAnalysis], default = None
+        parent Analysis
+    children: List[AbstractAnalysis], default []
```

Comment:
Should this be `default = []` to match others?

Reply:
fixed